### PR TITLE
cmd/swarm-smoke-pss: initial pss smoke tests

### PR DESCRIPTION
**Pull request by nonsense**

### CODE REVIEW

File: cmd/swarm-smoke-pss/main.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
+// Copyright 2018 The go-ethereum Authors
```
Comment (acud) [inline cmd/swarm-smoke-pss/main.go]: fix to swarm copy

**Comment by acud** [comment]
@nolash could you please have a second look at the code in this PR?

File: cmd/swarm-smoke-pss/main.go [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		cli.IntFlag{
+			Name:        "msgtimeout",
+			Value:       1,
+			Usage:       "timeout in seconds until a message is received",
+			Destination: &pssMessageTimeout,
+		},
+		cli.IntFlag{
+			Name:        "msgcount",
+			Value:       10,
+			Usage:       "number of pss messages that should be send in the pss smoke test",
```
Comment (nolash) [inline cmd/swarm-smoke-pss/main.go]: sen_t_

File: cmd/swarm-smoke-pss/main.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+		cli.IntFlag{
+			Name:        "timeout",
+			Value:       120,
+			Usage:       "timeout in seconds after which kill the process",
+			Destination: &timeout,
+		},
+		cli.IntFlag{
+			Name:        "msgtimeout",
+			Value:       1,
+			Usage:       "timeout in seconds until a message is received",
```
Comment (nolash) [inline cmd/swarm-smoke-pss/main.go]: time to wait before it fails?

File: cmd/swarm-smoke-pss/main.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+			Action: wrapCliCommand("sym", pssSymCheck),
+		},
+		{
+			Name:   "raw",
+			Usage:  "send and receive multiple raw messages across random nodes",
+			Action: wrapCliCommand("raw", pssRawCheck),
+		},
+		{
+			Name:   "all",
+			Usage:  "send and receive raw, sym and asym messages across random nodes",
```
Comment (nolash) [inline cmd/swarm-smoke-pss/main.go]: But in sequence, not mixed. Perhaps be specific about it.

File: cmd/swarm-smoke-pss/pss.go [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+			err = s.sendAsymMessage(j.sender, j.receiver, j.msg)
+			if err != nil {
+				log.Error("error sending asym message", "err", err)
+			}
+		default:
+			err = fmt.Errorf("invalid pssMode %d", mode)
+			log.Error("error processing job type", "err", err)
+		}
+
+		err = s.waitForMsg()
```
Comment (nolash) [inline cmd/swarm-smoke-pss/pss.go:271]: I assume the only thing we want to check here is correctness in the sense that messages are actually delivered, regardless of performance?

File: cmd/swarm-smoke-pss/pss.go [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+}
+
+// waitForMsg blocks until a msg is received or a timeout is reached
+func (s *pssSession) waitForMsg() error {
+	select {
+	case res := <-s.msgC:
+		resIdx := toMsgIdx(res.Msg)
+		resJob, ok := s.jobs[resIdx]
+		if !ok {
+			return fmt.Errorf("corrupt message for job %s", resIdx)
```
Comment (nolash) [inline cmd/swarm-smoke-pss/pss.go]: Actually it's not `job.resIdx` thats corrupt but it does not exist?

File: cmd/swarm-smoke-pss/pss.go [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+	c, err := rand.Read(symkey)
+	if err != nil {
+		return err
+	} else if c < 32 {
+		return fmt.Errorf("symkey size mismatch, expected 32 got %d", c)
+	}
+
+	// set the secret on both nodes
+	var senderSymKeyID string
+	err = sender.client.Call(&senderSymKeyID, "pss_setSymmetricKey", symkey, s.topic, hexutil.Encode(receiver.addr), true)
```
Comment (nolash) [inline cmd/swarm-smoke-pss/pss.go]: Last param can be `false` here.

File: cmd/swarm-smoke-pss/main.go (hunk 4 above)
Comment (acud) [reply cmd/swarm-smoke-pss/main.go]: not sure if i understand but i changed the comment anyway

File: cmd/swarm-smoke-pss/pss.go (hunk 5 above)
Comment (acud) [reply cmd/swarm-smoke-pss/pss.go:271]: let's start with that, yes

**Comment by acud** [comment]
@nolash I picked this up so we can get this merged, comments have been addressed

**Review by nolash — APPROVED** [review]

**Review by acud — APPROVED** [review]

**Pull request merged**